README.md

```diff
@@ -345,6 +345,13 @@ TYK_PMP_PUMPS_MONGOAGG_TYPE=mongo-pump-aggregate
 TYK_PMP_PUMPS_MONGOAGG_META_USEMIXEDCOLLECTION=true
 ```
 
+###### Self Healing
+The default max size of a document in MongoDB is 16Mb. In Mongo Aggregate, when we try to update a document, and the size reaches the limit, an error is received.
+When Self Healing is enabled and a document size hits 16Mb, Pump will create a new document with the current timestamp and reduce the aggregation time configuration by half. So, if the current_aggregation time is equal to 50, it will be reduced to 25. Also, the data that was being written, is sent to the new document created using its timestamp. 
```

Comment:
The Mongo Aggregate pump creates a new document in the database for each "aggregation period"; the length of that period is defined by `current_aggregation_time`. If, during that period (in minutes) the document grows beyond 16MB, the error will be received and no more records will be recorded until the end of the aggregation period (when a new document will be created).

The Self Healing option in the Mongo Aggregate Pump avoids this data loss by monitoring the size of the current document. When this hits the 16MB limit, Pump will automatically create a new document with the current timestamp and start writing to that instead.

When it does this, the Pump will halve the `current_aggregation_time ` so that it aggregates records for half the time period originally configured before creating a new document, reducing the risk of repeatedly hitting the maximum document size.

This self healing is repeatable, however, such that if the document size does reach maximum (16MB) even with the new shorter aggregation period, a new document will be created and the `current_aggregation_time` will be halved again.

The minimum value for `current_aggregation_time` is 1; Self Healing cannot reduce it beyond this value.